apps/api/v2/src/ee/schedules/services/response/response.service.ts

```diff
@@ -0,0 +1,68 @@
+import { SchedulesRepository } from "@/ee/schedules/schedules.repository";
+import { ScheduleResponse, schemaScheduleResponse } from "@/ee/schedules/services/response/zod/response";
+import { Injectable } from "@nestjs/common";
+import { User } from "@prisma/client";
+
+import {
+  transformWorkingHoursForClient,
+  transformAvailabilityForClient,
+  transformDateOverridesForClient,
+} from "@calcom/platform-libraries";
+import type {
+  ScheduleWithAvailabilities,
+  ScheduleWithAvailabilitiesForWeb,
+} from "@calcom/platform-libraries";
+
+@Injectable()
+export class ResponseService {
+  constructor(private readonly schedulesRepository: SchedulesRepository) {}
+
+  async formatSchedule(
+    forAtom: boolean,
+    user: User,
+    schedule: ScheduleWithAvailabilities
+  ): Promise<ScheduleResponse | ScheduleWithAvailabilitiesForWeb> {
+    if (forAtom) {
+      const usersSchedulesCount = await this.schedulesRepository.getUserSchedulesCount(user.id);
+      return this.transformScheduleForAtom(schedule, usersSchedulesCount, user);
+    }
+
+    return schemaScheduleResponse.parse(schedule);
+  }
+
+  async formatSchedules(
+    forAtom: boolean,
+    user: User,
+    schedules: ScheduleWithAvailabilities[]
+  ): Promise<ScheduleResponse[] | ScheduleWithAvailabilitiesForWeb[]> {
+    if (forAtom) {
+      const usersSchedulesCount = await this.schedulesRepository.getUserSchedulesCount(user.id);
+      return Promise.all(
+        schedules.map((schedule) => this.transformScheduleForAtom(schedule, usersSchedulesCount, user))
+      );
+    }
+    return schedules.map((schedule) => schemaScheduleResponse.parse(schedule));
+  }
```

Comment:
Why not keep the `forAtom` parameter here optional and instead give it a default value of false? That way only when we want to fetch schedule for any atom we'll pass that param.

Reply:
It is not optional because it will always be defined - the controller (endpoint) uses "@ForAtom" decorator that always returns true or false, which then is passed to formatter.